bulldozer/merge.go

```diff
@@ -195,3 +197,30 @@ func allCommits(ctx context.Context, pullCtx pull.Context, client *github.Client
 
 	return repositoryCommits, nil
 }
+
+func postCommentIfUserActionable(ctx context.Context, client *github.Client, owner, repo string, prNum int, githubError github.ErrorResponse) {
+	commentBody := "Error occurred when attempting merge: " + githubError.Message
+
+	switch {
+	case strings.HasPrefix(githubError.Message, "You're not authorized to push to this branch."):
+		commentBody = commentBody + "\n\nThe bulldozer bot account may not be allowed to merge the branch."
+
+	case githubError.Message == "Merge commits are not allowed on this repository.":
```

Comment:
I think this one could also be something along the lines of `Squash and merge is not allowed` or `Rebase is not allowed`.

Reply:
yea, this sort of logic is best captured in the abstraction you described above.